src/crypto/openssl/key_pair.cpp

```diff
@@ -202,6 +202,7 @@ namespace ccf::crypto
   {
     Unique_EVP_PKEY_CTX pctx(key);
     OpenSSL::CHECK1(EVP_PKEY_sign_init(pctx));
+    OpenSSL::CHECK1(EVP_PKEY_CTX_set_rsa_padding(pctx, RSA_PKCS1_PSS_PADDING));
```

Comment:
This fails `crypto_test` ...

Reply:
Do we know why?

Reply:
```
11: ../src/crypto/test/crypto.cpp:111:
11: TEST CASE:  Sign, fail to verify with bad signature
11:
11: ../src/crypto/test/crypto.cpp:111: ERROR: test case THREW exception: OpenSSL error: error:0D07803A:asn1 encoding routines:asn1_item_embed_d2i:nested asn1 error
11:   logged: With curve: 0x5653e7b671e9
```

In a nutshell, the second option fails signing on setting padding
```
static constexpr CurveID supported_curves[] = {
  CurveID::SECP384R1,  // OK
  CurveID::SECP256R1,  // FAILS!
  CurveID::SECP256K1}; // Skipped
```

Reply:
That looks like EC signing, not RSA?

Reply:
Yeap, I spotted that after learning what key types are. Now it passes, 'cause I only set padding to RSA keys